src/components/Drawer.js

```diff
@@ -0,0 +1,146 @@
+const React = require('react');
+const Velocity = require('velocity-animate');
+
+const StyleConstants = require('../constants/Style');
+
+const Icon = require('../components/Icon');
+
+const Drawer = React.createClass({
+  propTypes: {
+    duration: React.PropTypes.number,
+    easing: React.PropTypes.array,
+    isOpen: React.PropTypes.bool,
```

Comment:
@bsbeeks, @cerinman I've got all the transitions working succesfully on mount, but the unmount is posing a problem I've been working through unsuccesfully. I'm going to do some research on this and try to come up with a good solution. Right now the React Animations add on is looking the most hopeful. 

https://facebook.github.io/react/docs/animation.html